.github/workflows/provider-release.yaml

```diff
@@ -0,0 +1,38 @@
+# Terraform Provider release workflow.
+name: Release
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+permissions: {}
+
+jobs:
+  goreleaser:
+    permissions:
+      # GitHub considers creating releases and uploading assets as writing contents.
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@8v3
+        with:
+          fetch-depth: 0
```

Comment:
do we need this fetch depth? i always am confused when we need it

Reply:
Yeah we do (https://goreleaser.com/ci/actions/#fetch-depthness) I think it's so that the workflow has the info it needs to create an actual release from the GHA run

Reply:
ok gr8